index.js

```diff
@@ -143,42 +190,61 @@ const line = []
 // *** Uncomment the lines below to test
 // console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+let lineNumber = 0;
+
+function takeATicketNumber(arr) {
+  lineNumber += 1
+  arr.push(lineNumber);
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+  return `Welcome. You are ticket number ${lineNumber}`
+}
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc () {
+  let newNum = 0;
+  return function (arr) {
```

Comment:
Great solution!
Remember that JS has first-class functions, so you can return function definitions from a function and you can include variables that's in scope to the body of that function!